### PR TITLE
Fix iOS touch drag behavior

### PR DESCRIPTION
**Pull request by Renzo-Olivares**
Before this change on a quick touch drag the cursor, where the touch is not on the previous collapsed selection, the cursor would move to the tapped position.

After this change on a quick touch drag the cursor does not move unless the touch is on the previously collapsed selection. This is inline with native behavior.

Before|After|Native
--|--|--
<video src="https://user-images.githubusercontent.com/948037/233224775-f33b42b5-5638-416c-9278-39ecd964e3bb.mov" />|<video src="https://user-images.githubusercontent.com/948037/233224760-2d1af657-8d99-45fc-8499-9567f17d533e.mov" />|<video src="https://user-images.githubusercontent.com/948037/233224790-f5997cfa-7370-4891-8952-11ef8057a729.mov" />

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

File: packages/flutter/lib/src/widgets/text_selection.dart
Comment (justinmc) [inline packages/flutter/lib/src/widgets/text_selection.dart:2617]: Why didn't the magnifier show up in the video in your description?

File: packages/flutter/lib/src/widgets/text_selection.dart
Comment (justinmc) [inline packages/flutter/lib/src/widgets/text_selection.dart:2575]: Do you still need the `??` even though this is now in the `Start` handler?

**Review by justinmc — APPROVED** [review]
LGTM 👍

File: packages/flutter/lib/src/widgets/text_selection.dart
Comment (Renzo-Olivares) [reply packages/flutter/lib/src/widgets/text_selection.dart:2617]: I took the videos before the magnifier was merged in, it works though.

Comment (justinmc) [reply packages/flutter/lib/src/widgets/text_selection.dart:2617]: Ok cool 👍

File: packages/flutter/lib/src/widgets/text_selection.dart
Comment (Renzo-Olivares) [reply packages/flutter/lib/src/widgets/text_selection.dart:2575]: Good point i'll remove it.

**Comment by Renzo-Olivares** [comment]
This PR is failing because `_showMagnifierIfSupportedByPlatform` and its `hide` variant have not been merged into g3 yet. I'll merge once that happens.

**Pull request merged**
